controllers/iochaos/podiochaosmanager/podiochaosmanager.go

```diff
@@ -60,8 +70,16 @@ func (m *PodIoManager) WithInit(key types.NamespacedName) *PodIoTransaction {
 	return t
 }
 
+// KeyErrorTuple is a tuple (Key, Err)
+type KeyErrorTuple struct {
+	Key types.NamespacedName
+	Err error
+}
+
 // Commit will update all modifications to the cluster
-func (m *PodIoManager) Commit(ctx context.Context) error {
+func (m *PodIoManager) Commit(ctx context.Context) chan KeyErrorTuple {
```

Comment:
What's the advantage about using a channel here?

 I think return a `[]error` will be better, it still make sure method `Commit` will pending until all operation finished.

Reply:
If we use chan here, we can add a buffer for it.

Reply:
@STRRL The way to create a `[]KeyErrorTuple` is just by creating a channel and send the `KeyErrorTuple` into this channel, and the receiver could append these tuples to the slice.

Then why not return the channel directly?

Reply:
For keeping it simple and stupid.

There are 2 reasons that I considered:
- return `[]KeyErrorTuple` means `Commit()` is a synchronized method, we do not need to think about anything about multi-thread. Once channel is used, user should think about "Should I close it?", "Are there any other consumers also read elements from this channel?". Also exposing a channel means here is a possibility that receiver codes could close it(which is dangerous, sending item into a closed channel and closing a closed channel will make panic), and the scope of parallelism is escape from `Commit()` to its caller,  it's a bad-smell about parallelism pattern.
- Using `[]KeyErrorTuple` will not make perform decrease. A for-loop still perform as single-thread.

Reply:
@STRRL I prefer to make it a read-only channel

@cwen0 Do you have any suggestions on the buffer size? I think blocking at the end of execution is not a problem (as the rest part is all about calculation, without IO operation)

Reply:
`Commit(ctx) <-chan` works like "async-commit", since our use-case is quite sample, I think it is acceptable.

PS: If we have many around things need call this, I still suggest provide a "SyncCommit()" 🤪

Reply:
TBH, I also get confused when I see the `Commit` return is a channel, as @STRRL said, return channel means that it's an async operation, and I think we don't need to return the channel to outside caller, XXXManager calls `m.Commit` and even doesn't have the requirements for async communication, e.g. abort the job.

See the [errgroup test example](https://go.googlesource.com/sync/+/master/errgroup/errgroup_test.go#65), it's similar to this case, and I suggest follow the standard code style.

Reply:
@Colstuwjx Nice example! I haven't realized that I can modify a slice if the capacity is big enough 😿 , so no need for a channel.

Reply:
Fixed